s3/s3etag/s3etag.go

```diff
@@ -0,0 +1,141 @@
+// Package s3etag implements the AWS ETag hash as described in
+//
+// http://permalink.gmane.org/gmane.comp.file-systems.s3.s3tools/583
```

Comment:
I have a slight concern that this isn't officially documented by Amazon.  All [AWS says about it](http://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonResponseHeaders.html) is

> If an object is created by either the Multipart Upload or Part Copy operation, the ETag is not an MD5 digest, regardless of the method of encryption.

Reply:
Agreed, AWS is pretty quiet on the ETag.  I share your concern.  Still, the ETag as implemented works and addresses a missing use case: upload large files to S3 and (crypt) check them.